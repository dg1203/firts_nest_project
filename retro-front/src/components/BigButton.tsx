import styled, {keyframes} from 'styled-components';

interface BigButtonInterface {
  isLoading?: boolean;
  isDisabled?: boolean;
}

const loader = keyframes`
  0%{ width : 10px; height: 10px; opacity: 1; }
  33%{ width : 20px; height: 20px; opacity: .5; }
  33%{ width : 30px; height: 30px; opacity: 0;}
  100%{ width : 10px; height: 10px; opacity: 1; }
`;

const loader1 = keyframes`
  0%{ width : 30px; height: 30px; opacity: 1; }
  33%{ width : 20px; height: 20px; opacity: .5; }
  33%{ width : 10px; height: 10px; opacity: 0;}
  100%{ width : 30px; height: 30px; opacity: 1; }
`;

const BigButton = styled.button<BigButtonInterface>`
  border-radius: 5px;
  background: ${props => props.theme.colors.blue};
  color: white;
  font-size: 1.5rem;
  width: 100%;
  max-width: 500px;
  height: 54px;
  border: none;
  cursor: pointer;
  transition: 0.3s;
  position: relative;
  opacity: ${props => props.isDisabled ? 0.5 : 1};
  &:hover {
    opacity: 0.8;
  }
  &:after {
    content: '';
    border-radius: 50%;
    position: absolute;
    height: 10px;
    border: 2px solid ${props => props.theme.colors.white};
    margin: auto;
    left: 0;
    bottom: 0;
    right: 0;
    top: 0;
    width: 10px;
    animation: ${loader} 1s linear infinite;
    display: ${props => (props.isLoading ? 'block' : 'none')};
  }
  &:before {
    content: '';
    border-radius: 50%;
    position: absolute;
    height: 30px;
    border: 2px solid ${props => props.theme.colors.white};
    margin: auto;
    left: 0;
    bottom: 0;
    right: 0;
    top: 0;
    width: 30px;
    animation: ${loader1} 1s linear infinite;
    display: ${props => (props.isLoading ? 'block' : 'none')};
  }
  @media screen and (max-width: 768px) {
    max-width: 100%;
    width: 100%;
  }
`;

export default BigButton;
