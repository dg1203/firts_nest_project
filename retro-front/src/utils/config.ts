import axios from 'axios';

export const setAxiosConfig = (accessToken: string) => {
  axios.defaults.headers.common['Authorization'] = `Bearer ${accessToken}`;
  localStorage.setItem('accessToken', accessToken);
};
