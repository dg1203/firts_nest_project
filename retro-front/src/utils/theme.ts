interface ColorsInterface {
  blue: string;
  grey: string;
  white: string;
  red: string;
  black: string;
  green: string;
}

interface ThemeInterface {
  colors: ColorsInterface;
}

export const theme: ThemeInterface = {
  colors: {
    blue: '#2980b9',
    grey: '#bdc3c7',
    white: '#ffffff',
    red: '#c0392b',
    black: '#34495e',
    green: '#27ae60'
  }
};
