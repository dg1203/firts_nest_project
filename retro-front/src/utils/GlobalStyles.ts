import { createGlobalStyle } from "styled-components";
import { theme } from "./theme";

const GlobalStyles = createGlobalStyle`
  @import url(â€˜https://fonts.googleapis.com/css?family=Montserrat|Roboto');
  * {
    box-sizing: border-box;
  }
  body {
    margin: 0px;
    padding: 0px;
    font-family: Roboto, sans-serif;
    overflow-x: hidden;
  }
  a {
    color: ${theme.colors.blue};
    text-decoration: none;
  }
  form {
    width: 100%;
  }
`;

export default GlobalStyles;