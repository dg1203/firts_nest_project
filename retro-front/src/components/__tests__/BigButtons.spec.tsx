import React from 'react';
import BigButton from '../BigButton';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from '../../utils/theme';

const renderComponent = (props: any) =>
  render(
    <ThemeProvider theme={theme}>
      <BigButton {...props} />
    </ThemeProvider>
  );

describe('<BigButton />', () => {
  it('To match snapshot', () => {
    const { asFragment } = renderComponent({});
    expect(asFragment()).toMatchSnapshot();
  });
});
