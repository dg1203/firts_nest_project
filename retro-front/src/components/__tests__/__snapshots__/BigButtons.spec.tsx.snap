// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BigButton /> To match snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-AxjAm ckPEsE"
  />
</DocumentFragment>
`;
